src/core/PlaneWaveBasis.jl

```diff
@@ -0,0 +1,216 @@
+using FFTW
+
+@doc raw"""
+    PlaneWaveBasis(lattice::SMatrix{3, 3, T}, grid_size::SVector{3, I},
+                   Ecut::Number, kpoints, kweights) where {T <: Real, I <: Integer}
+
+Create a plane-wave basis.
+
+## Examples
+```julia-repl
+julia> b = PlaneWaveBasis(TODO)
+```
+
+## Arguments and struct fields
+- `lattice`: Real-space lattice vectors in columns
+- `recip_lattice`: Reciprocal lattice vectors in columns
+- `grid_size`:     Size of the rectangular Fourier grid used as the
+                   density basis ``B_ρ``. In each dimension `idim` the
+                   range of wave vectors (in integer coordinates) extends from
+                   `-ceil(Int, (grid_size[idim]-1) / 2)` up to
+                   `floor(Int, (grid_size[idim]-1) / 2)`. No optimisation is done
+                   on the size of the grid with respect to performing FFTs.
+- `Ecut`:          Kinetic energy cutoff in Hartree
+- `kpoints`:       List of ``k``-Points in fractional coordinats
+- `kweights`:      List of corresponding weights for the Brillouin-zone integration.
+- `kbasis`:        List of wave vectors (in integer coordinates) for each ``k``-Point,
+                   chosen such that ``|G + k|^2/2 \leq Ecut`` where ``G`` is the
```

Comment:
Technically nothing in the PlaneWaveBasis structure assumes this, and the rest of the code works fine without this assumption (which you could reasonably want to change to have a different basis selection). So I'd leave that out of the struct, and document this in the convenience constructors.

Reply:
Done.